File: pkg/types/events.go
package types

type ExecEvent struct {
	Type       string `json:"type"`
	Timestamp  int64  `json:"timestamp"`
	PID        uint32 `json:"pid"`
	PPID       uint32 `json:"ppid"`
	CgroupID   string `json:"cgroupId"`
	Comm       string `json:"comm"`
	ParentComm string `json:"parentComm"`
	Blocked    bool   `json:"blocked"`
}

type FileEvent struct {
	Type      string `json:"type"`
	Timestamp int64  `json:"timestamp"`
	PID       uint32 `json:"pid"`
	CgroupID  string `json:"cgroupId"`
	Flags     uint32 `json:"flags"`
	Filename  string `json:"filename"`
	Blocked   bool   `json:"blocked"`
}

type ConnectEvent struct {
	Type      string `json:"type"`
	Timestamp int64  `json:"timestamp"`
	PID       uint32 `json:"pid"`
	CgroupID  string `json:"cgroupId"`
	Family    uint16 `json:"family"`
	Port      uint16 `json:"port"`
	Addr      string `json:"addr"`
	Blocked   bool   `json:"blocked"`
}

type Alert struct {
	ID          string `json:"id"`
	Timestamp   int64  `json:"timestamp"`
	Severity    string `json:"severity"`
	RuleName    string `json:"ruleName"`
	Description string `json:"description"`
	PID         uint32 `json:"pid"`
	ProcessName string `json:"processName"`
	ParentName  string `json:"parentName"`
	CgroupID    string `json:"cgroupId"`
	Action      string `json:"action"`
	Blocked     bool   `json:"blocked"`
}

type Workload struct {
	ID           string `json:"id"`
	CgroupPath   string `json:"cgroupPath"`
	ExecCount    int64  `json:"execCount"`
	FileCount    int64  `json:"fileCount"`
	ConnectCount int64  `json:"connectCount"`
	AlertCount   int64  `json:"alertCount"`
	BlockedCount int64  `json:"blockedCount"`
	FirstSeen    int64  `json:"firstSeen"`
	LastSeen     int64  `json:"lastSeen"`
}

type ProcessInfo struct {
	PID       uint32 `json:"pid"`
	PPID      uint32 `json:"ppid"`
	Comm      string `json:"comm"`
	CgroupID  string `json:"cgroupId"`
	Timestamp int64  `json:"timestamp"`
}

File: frontend/src/types/events.ts
// Event Types - Phase 4

export type EventType = 'exec' | 'file' | 'connect'

export interface EventHeader {
  timestamp: number
  pid: number
  ppid?: number
  cgroupId: string
  uid?: number
  gid?: number
  comm: string
}

export interface ExecEvent {
  id: string
  type: 'exec'
  header: EventHeader
  parentComm: string
  filename: string
  argv0?: string
  blocked?: boolean
}

export interface FileEvent {
  id: string
  type: 'file'
  header: EventHeader
  filename: string
  flags: number
  ino?: number
  dev?: number
  blocked?: boolean
}

export interface ConnectEvent {
  id: string
  type: 'connect'
  header: EventHeader
  family: number
  port: number
  addr: string
  blocked?: boolean
}

export type SecurityEvent = ExecEvent | FileEvent | ConnectEvent

export interface QueryFilter {
  types?: EventType[]
  processes?: string[]
  actions?: string[]
  pids?: number[]
  cgroupIds?: string[]
  timeWindow?: {
    start: string
    end: string
  }
  correlation?: boolean
}

export interface QueryRequest {
  filter?: QueryFilter
  semantic?: string
  page?: number
  limit?: number
  sortBy?: 'time' | 'relevance'
  sortOrder?: 'asc' | 'desc'
}

export interface QueryResponse {
  events: SecurityEvent[]
  total: number
  page: number
  limit: number
  totalPages: number
  type_counts?: {
    exec: number
    file: number
    connect: number
  }
}

